libs/velo-external-db-core/src/web/jwt-auth-middleware.spec.ts

```diff
@@ -0,0 +1,89 @@
+
+    const env = {
+        auth: Uninitialized,
+    }
+
+    beforeEach(() => {
+        ctx.externalDatabaseId = chance.word()
+        ctx.metasite = chance.word()
+        ctx.allowedMetasites = ctx.metasite
+        ctx.next = jest.fn().mockName('next')
+        env.auth = new JwtAuthenticator(ctx.externalDatabaseId, ctx.allowedMetasites, new WixDataFacadeMock(decodeBase64(ctx.authPublicKey))).authorizeJwt()
+    })
+
+    test('should authorize when JWT valid', async() => {
+        const token = jwt.sign({ iss: TOKEN_ISSUER, metasite: ctx.metasite }, decodeBase64(ctx.authPrivateKey), { algorithm: 'RS256' })
+        await env.auth(driver.requestBodyWith(ctx.ownerRole, ctx.dataPath, `Bearer ${token}`), Uninitialized, ctx.next)
+
+        expect(ctx.next).toHaveBeenCalledWith()
+    })
+
+    test('should authorize when JWT valid, only with second public key', async() => {
+        const token = jwt.sign({ iss: TOKEN_ISSUER, metasite: ctx.metasite }, decodeBase64(ctx.authPrivateKey), { algorithm: 'RS256' })
+        env.auth = new JwtAuthenticator(ctx.externalDatabaseId, ctx.allowedMetasites, new WixDataFacadeMock(decodeBase64(ctx.otherAuthPublicKey), decodeBase64(ctx.authPublicKey))).authorizeJwt()
+        await env.auth(driver.requestBodyWith(ctx.ownerRole, ctx.dataPath, `Bearer ${token}`), Uninitialized, ctx.next)
+
+        expect(ctx.next).toHaveBeenCalledWith()
+    })
+
+    test('should throw when JWT metasite is not allowed', async() => {
+        const token = jwt.sign({ iss: TOKEN_ISSUER, metasite: chance.word() }, decodeBase64(ctx.authPrivateKey), { algorithm: 'RS256' })
+        await env.auth(driver.requestBodyWith(ctx.ownerRole, ctx.dataPath, `Bearer ${token}`), Uninitialized, ctx.next)
+
+        expect(ctx.next).toHaveBeenCalledWith(new UnauthorizedError('You are not authorized'))
+    })
+
+    test('should throw when JWT has no metasite claim', async() => {
+        const token = jwt.sign({ iss: TOKEN_ISSUER }, decodeBase64(ctx.authPrivateKey), { algorithm: 'RS256' })
+        await env.auth(driver.requestBodyWith(ctx.ownerRole, ctx.dataPath, `Bearer ${token}`), Uninitialized, ctx.next)
+
+        expect(ctx.next).toHaveBeenCalledWith(new UnauthorizedError('You are not authorized'))
+    })
+
+    test('should throw when JWT issuer is not Wix-Data', async() => {
+        const token = jwt.sign({ iss: chance.word(), metasite: ctx.metasite }, decodeBase64(ctx.authPrivateKey), { algorithm: 'RS256' })
+        await env.auth(driver.requestBodyWith(ctx.ownerRole, ctx.dataPath, `Bearer ${token}`), Uninitialized, ctx.next)
+
+        expect(ctx.next).toHaveBeenCalledWith(new UnauthorizedError('You are not authorized'))
+    })
+
+    test('should throw when JWT has no issuer', async() => {
+        const token = jwt.sign({ metasite: ctx.metasite }, decodeBase64(ctx.authPrivateKey), { algorithm: 'RS256' })
+        await env.auth(driver.requestBodyWith(ctx.ownerRole, ctx.dataPath, `Bearer ${token}`), Uninitialized, ctx.next)
```

Comment:
I think it would be clearer to send here undefined/null instead of Uninitialized